torchvision/prototype/datasets/utils/_dataset.py

```diff
@@ -181,3 +182,41 @@ def load(
 
     def _generate_categories(self, root: pathlib.Path) -> Sequence[Union[str, Sequence[str]]]:
         raise NotImplementedError
+
+
+class Dataset2(IterDataPipe[Dict[str, Any]], abc.ABC):
+    @staticmethod
+    def _verify_str_arg(
```

Comment:
Unless we have a use case for `verify_str_arg` outside of our datasets, I would simply move the functionality into this method later. I just didn't do it here to avoid copy-pasting the code.

Even if we move `verify_str_arg` to say `torch.utils._internal`, I still think it is ok to have it as a method on the dataset class. We already have quite a few imports at the top of each dataset implementation. We probably don't need even more.

Reply:
Moving it as a method and removing it from utils sounds good.

I wouldn't worry about the number of imports personally